doc/source/ray-core/runtime_env_auth.md

```diff
@@ -0,0 +1,120 @@
+(runtime-env-auth)=
+# Authenticating Remote URIs in your runtime_env
+
+This section helps you:
+
+* Avoid leaking remote URI credentials in your `runtime_env`
+* provide credentials safely in KubeRay
+* understand best practices when authenticating your remote URI
+
+## Authenticating Remote URIs
+
+You can add dependencies to your `runtime_env` with [remote URIs](remote-uris). This is straightfoward for files hosted publicly, since you can simply paste the public URI into your `runtime_env`:
+
+```python
+runtime_env = {"working_dir": (
+        "https://github.com/"
+        "username/repo/archive/refs/heads/master.zip"
+    )
+}
+```
+
+However, dependencies hosted privately, in a private GitHub repo for example, require authentication. One common way to authenticate is to insert credentials into the URI itself:
+
+```python
+runtime_env = {"working_dir": (
+        "https://username:personal_access_token@github.com/"
+        "username/repo/archive/refs/heads/master.zip"
+    )
+}
+```
+
+In this example, `personal_access_token` is a secret credential that authenticates this URI. While Ray can successfully access your dependencies using authenticated URIs, **you should not include secret credentials in your URIs** for two reasons:
+
+1. Ray may log the URIs used in your `runtime_env`, which means the Ray logs could contain your credentials.
+2. Ray stores your remote dependency package in a local directory, and it uses a parsed version of the remote URI– including your credential– as the directory's name.
+
+In short, your remote URI is not treated as a secret, so it should not contain secret info. Instead you should use a `netrc` file.
+
+## Running on VMs: the netrc File
+
+The [netrc file](https://www.gnu.org/software/inetutils/manual/html_node/The-_002enetrc-file.html) contains credentials that Ray uses to automatically log into remote servers. You can set your credentials in this file instead of in the remote URI:
```

Comment:
```suggestion
The [netrc file](https://www.gnu.org/software/inetutils/manual/html_node/The-_002enetrc-file.html) contains credentials that Ray uses to automatically log into remote servers. Set your credentials in this file instead of in the remote URI:
```